messaging/kafka/src/main/java/io/helidon/messaging/kafka/SimpleKafkaConsumer.java

```diff
@@ -0,0 +1,322 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.messaging.kafka;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.common.context.Context;
+import io.helidon.common.context.Contexts;
+import io.helidon.config.Config;
+import io.helidon.messaging.kafka.connector.KafkaMessage;
+import io.helidon.messaging.kafka.connector.SimplePublisher;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.errors.WakeupException;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * Simple Kafka consumer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}
+ * <p>
+ * Usage:
+ * <pre>{@code
+ *   try (SimpleKafkaConsumer<Long, String> c = new SimpleKafkaConsumer<>("test-channel", Config.create())) {
+ *         c.consumeAsync(r -> System.out.println(r.value()));
+ *   }
+ * }</pre>
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+public class SimpleKafkaConsumer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(SimpleKafkaConsumer.class.getName());
+    private final KafkaConfigProperties properties;
+
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+    private final PartitionsAssignedLatch partitionsAssignedLatch = new PartitionsAssignedLatch();
+    private final String consumerId;
+    private ExecutorService executorService;
```

Comment:
This seems to be mutable (and wrongly).
If the consumer is only for a single use, make sure you construct the instance with all the configuration, then have a method to start listening. This will simplify a lot of checks.
Also not sure why there is `executorService` and `externalExecutorService`

Reply:
Now it is simpler.

Regarding the executors, I followed your suggestion about usage of one unique ScheduledExecutorService that is shared between all the kafka consumers.

That reduces the overload, but it increases the complexity of the class because:
- KafkaConsumer.close cannot be invoked in the task execution (because there is no while loop anymore).
- Now it is multi thread and KafkaConsumer is not multi thread. So we need to make sure only one thread is using it every time.